recipes/libxml2/all/conanfile.py

```diff
@@ -63,8 +63,9 @@ class Libxml2Conan(ConanFile):
     options = {name: [True, False] for name in default_options.keys()}
 
     @property
-    def _option_names(self):
-        return [name for name in self.info.options.keys() if name not in ["shared", "fPIC", "include_utils"]]
+    def _configure_option_names(self):
+        return [name for name in self.default_options.keys() if (name in self.options)
```

Comment:
Yeah, things like this is why we need to review its API for 2.0, will let you know once we find something we're happy with :)

Reply:
We have decided that this is the way to go for now. As for your specific qustion:
`default_options` has a key because there's not much preprocessing needed for those, so they stay as their original dict all the way thru. On the other hand, options is heavily modified, so we would need to implement keys() by ourselves. 
The main reason why this is not done is because the meaning of such a method would not be clear. Would calling keys() return the options for the current recipe only? Or would it make sense for it to return options defined upstream in the dependency the graph? As we couldn't find a reason for why any of those would be better (and less confusing/surprising for users), we decided not to expose such a functionality.

This does not mean why're hard against having something like this! We just need to think better about it, but in the meanwhile we can fix this recipe for v2 :)